pytradfri/device.py

```diff
@@ -223,8 +215,38 @@ def set_predefined_color(self, colorname, *, index=0):
             raise ColorError('Invalid color specified: %s',
                              colorname)
 
-    def set_rgb_color(self, r, g, b, *, index=0):
-        return self.set_values(rgb_to_xyY(r, g, b), index=index)
+    @property
+    def _kelvin_range(self):
```

Comment:
Can we not predetermine most of these when we initialise the light?

Reply:
Not quite sure I follow your thougts here?

Reply:
We could construct this data in an  `__init__`?

Reply:
Do you mean to extend __init__ as so?
```
    def __init__(self, device):
        self._device = device

        if not self.can_set_kelvin:
            # White bulb
            self._kelvin_range = (None, None)
        if not self.can_set_color:
            # White spectrum bulb
            self._kelvin_range = (MIN_KELVIN_WS, MAX_KELVIN_WS)
        # Color bulb
        self._kelvin_range = (MIN_KELVIN, MAX_KELVIN)


```

Reply:
Something like that, yeah. @property is useful if something is going to change between calls.

Reply:
I've made an attempt to move a lot of stuff into __init__ now